tools/individual-pkg-gen/copy-files-removing-deps.ts

```diff
@@ -0,0 +1,201 @@
+      const destination = path.join(destDir, sourceFileName);
+
+      let fileProcessed = false;
+      if (sourceFileName === 'package.json') {
+        const phase1PackageJson = fs.readJsonSync(source);
+        if (phase1PackageJson.name === pkg.name) {
+          const pkgUnscopedName = `${pkg.name.substring('@aws-cdk/'.length)}`;
+
+          phase1PackageJson.name = `@aws-cdk-lib-alpha/${pkgUnscopedName}`;
+          phase1PackageJson.repository.directory = `packages/individual-packages/${pkgUnscopedName}`;
+
+          // disable awslint (some rules are hard-coded to @aws-cdk/core)
+          phase1PackageJson.awslint = {
+            exclude: ['*:*'],
+          };
+
+          // add a pkglint exemption for the 'package name = dir name' rule
+          const pkglint = phase1PackageJson.pkglint || {};
+          pkglint.exclude = [
+            ...(pkglint.exclude || []),
+            'naming/package-matches-directory',
+            // the experimental packages need the "real" assert dependency
+            'assert/assert-dependency',
+          ];
+          phase1PackageJson.pkglint = pkglint;
+
+          // turn off the L1 generation, which uses @aws-cdk/ modules
+          if (phase1PackageJson.scripts?.gen === 'cfn2ts') {
+            delete phase1PackageJson.scripts.gen;
+          }
+
+          /* ****** handle dependencies ****** */
+
+          // regular dependencies
+          const alphaDependencies: { [dep: string]: string } = {};
+          for (const dependency of Object.keys(phase1PackageJson.dependencies || {})) {
+            // all 'regular' dependencies on alpha modules will be converted to
+            // a pair of devDependency on '0.0.0' and peerDependency on '^0.0.0',
+            // and the package will have no regular dependencies anymore
+            if (alphaPackages[dependency]) {
+              alphaDependencies[alphaPackages[dependency]] = pkg.version;
+            }
+          }
+          phase1PackageJson.dependencies = undefined;
+
+          const finalPackageJson = { ...phase1PackageJson };
+          // devDependencies
+          const alphaDevDependencies: { [dep: string]: string } = {
+            // we need jsii-rosetta in the dependencies,
+            // otherwise the binary cannot be found when running the 'extract' script
+            'jsii-rosetta': jsiiRosettaVersion,
+          };
+          const devDependencies = phase1PackageJson.devDependencies || {};
+          for (const devDependency of Object.keys(devDependencies)) {
+            if (devDependency.startsWith('@aws-cdk/')) {
+              delete devDependencies[devDependency];
+            }
+            if (alphaPackages[devDependency]) {
+              alphaDevDependencies[alphaPackages[devDependency]] = pkg.version;
+            }
+          }
+          devDependencies['@aws-cdk/assert'] = pkg.version;
+          devDependencies['aws-cdk-lib'] = pkg.version;
+          devDependencies.constructs = '^10.0.0';
```

Comment:
same here. As a general rule, we should check if the package depends (regular, dev or peer) on 'x', and only if it does, should its version be updated.

Reply:
Done.